### DISCUSSION TITLE
Looking for an IM expert to assist with PDF to PNG

### DISCUSSION THREAD

**Discussion by stockydale** [ImageMagick/ImageMagick#5145] [Consulting]
Hello,

We are looking for an IM expert to help us with finding the best parameters to convert PDF to PNG. We would like to take large technical PDF drawings and convert them to high res PNG images, with the minimum file size.

Currently we have found that the image conversion process used by Adobe Acrobrat DC is the best tool for the job, but this is a manual process. We would like to use IM (or similar) to automate this process to get a similar, pixel perfect output.

Looking for someone who has an in depth knowledge that can advise accordingly.

Thanks in advance

**Comment by urban-warrior** [comment]
Use the latest release of ImageMagick v7 and super-sample:

```
magick -density 400 image.pdf -resize 25% image.png
```

Since these are technical drawings, PNG compression should be adequate to return a minimum file size.  You can tweak the results by using color profiles, resizing in linear space, selecting the right resize filter (e.g., LanczosSharp), or choosing a specific PNG compression algorithm.  However, in most cases, you won't return significant benefit from any of these options.  The best visuals results are likely returned by super-sampling.

**Reply by stockydale** [reply]
![Comparison between Adobe and IM](https://user-images.githubusercontent.com/93651602/167862228-5a367260-c7ad-4087-b2b8-88697324ac73.png)

Thanks @urban-warrior - I have attached a sample of what we are achieving, versus what we would like to achieve. The bottom image is created using Adobe Acrobat DC. It is a lot clearer and therefore smaller in size - we would like to try and get the same output using IM.

**Reply by urban-warrior** [reply]
Can you post the original PDF so we can experiment?  Visually, it looks like you might want to use the LanczosSharp (or Lanczos2Sharp) resize filter to sharpen the results.

**Reply by stockydale** [reply]
Hi @urban-warrior do you have an email address I can send it to instead?

**Reply by urban-warrior** [reply]
We created a temp e-mail address, `d6qzzy@imagemagick.org`.

**Reply by stockydale** [reply]
Thanks @urban-warrior - I have just sent you an email.

**Reply by urban-warrior** [reply]
Got it.  We'll need a day or two to review.

**Reply by stockydale** [reply]
Thank you!

**Comment by urban-warrior** [comment]
ImageMagick utilizes the Ghostscript delegate library/program to render PDF.  Try using it directly:

```
gs -sDEVICE=png16m -r600 -dDownScaleFactor=3 -o stockydale.png stockydale.pdf
```

To remove the aliasing, try:

```
gs -sDEVICE=png16m -r300 -o stockydale.png stockydale.pdf
```

That's quite fast with a slightly smaller PNG result.  Or try these ImageMagick commands:

```
magick -density 300 stockydale.pdf -alpha remove -strip -interlace none -depth 8 stockydale-im.png
```

or for a smaller PNG:

```
magick-density 300 stockydale.pdf -alpha remove -strip -interlace none -depth 8 png8:stockydale.png
```

In terms of the quality of the rendering-- that's on Ghostscript.  It may render certain objects differently than Adobe.  Also, doesn't Adobe offer Cloud services to render PDF's, no manual intervention required?

**Reply by stockydale** [reply]
Thanks @urban-warrior we will give those a try.

Do you have a link for the Adobe cloud service you have mentioned?

**Reply by urban-warrior** [reply]
@fmw42 do you know if Adobe has batch conversion?  Otherwise, take a look at Appligent.  They have a batch processor.

**Reply by fmw42** [reply]
I do not know about batch processing by Adobe for PDF files. They could, but I just have no knowledge about it one way or the other.

I know little about Appligent other than it helps you process PDF files.
